lib/Shared.js

```diff
@@ -0,0 +1,197 @@
+var parseRange = require("range-parser");
+var pathIsAbsolute = require("path-is-absolute");
+var MemoryFileSystem = require("memory-fs");
+
+module.exports = function Shared(context) {
+	var share = {
+		setOptions: function(options) {
+			if(!options) options = {};
+			if(typeof options.watchOptions === "undefined") options.watchOptions = {};
+			if(typeof options.reporter !== "function") options.reporter = share.defaultReporter;
+			if(typeof options.log !== "function") options.log = console.log.bind(console);
+			if(typeof options.warn !== "function") options.warn = console.warn.bind(console);
+			if(typeof options.watchDelay !== "undefined") {
+				// TODO remove this in next major version
+				options.warn("options.watchDelay is deprecated: Use 'options.watchOptions.aggregateTimeout' instead");
+				options.watchOptions.aggregateTimeout = options.watchDelay;
+			}
+			if(typeof options.watchOptions.aggregateTimeout === "undefined") options.watchOptions.aggregateTimeout = 200;
+			if(typeof options.stats === "undefined") options.stats = {};
+			if(!options.stats.context) options.stats.context = process.cwd();
+			if(options.lazy) {
+				if(typeof options.filename === "string") {
+					var str = options.filename
+						.replace(/[\-\[\]\/\{\}\(\)\*\+\?\.\\\^\$\|]/g, "\\$&")
+						.replace(/\\\[[a-z]+\\\]/ig, ".+");
+					options.filename = new RegExp("^[\/]{0,1}" + str + "$");
+				}
+			}
+			context.options = options;
+		},
+		defaultReporter: function(reporterOptions) {
+			var state = reporterOptions.state;
+			var stats = reporterOptions.stats;
+			var options = reporterOptions.options;
+
+			if(state) {
+				var displayStats = (!options.quiet && options.stats !== false);
+				if(displayStats && !(stats.hasErrors() || stats.hasWarnings()) &&
+					options.noInfo)
+					displayStats = false;
+				if(displayStats) {
+					options.log(stats.toString(options.stats));
+				}
+				if(!options.noInfo && !options.quiet) {
+					options.log("webpack: bundle is now VALID.");
+				}
+			} else {
+				options.log("webpack: bundle is now INVALID.");
+			}
+		},
+		handleRangeHeaders: function handleRangeHeaders(content, req, res) {
```

Comment:
I added a comment in the method, suggesting a way forward. E.g. conditional(s) based on an object check for various APIs. A new context flag could also be added to indicate what API is being used. But I believe that can be addressed in a future PR for support of additional server(s).